src/dlgTriggerEditor.cpp

```diff
@@ -7233,6 +7235,24 @@ void dlgTriggerEditor::slot_closeSourceFind()
     mpSourceEditorEdbee->setFocus();
 }
 
+void dlgTriggerEditor::slot_sourceReplace()
+{
+    auto controller = mpSourceEditorEdbee->controller();
+    auto replaceText = mpSourceEditorFindArea->lineEdit_replaceText->text();
+    for (int i = 0; i < controller->textSelection()->rangeCount(); i++) {
+        auto &range = controller->textSelection()->range(i);
+        if (mpSourceEditorEdbee->textDocument()->text().mid(range.anchor(), range.length()) == replaceText) {
+            slot_sourceFindNext();
+            continue;
+        } else if (range.length() == 0) {
```

Comment:
:thinking: Isn't this something like the *else-after-return* antipattern - hitting the `continue`  means that the `else` is redundant? Also `((int) TextRange::length()) == 0` looks awfully like it can be replaced by `!TextRange::hasSelection()`:
```suggestion
        }
        if (!range.hasSelection()) {
```

Reply:
Ah, that's a leftover because I was originally using slot_sourceFindNext() in both branches and intended to just logically merge them until I realized I needed sourceFindPrevious in the length() == 0. I'll commit both of those changes, thanks!